Plugins/MaxLevel/MaxLevel.cpp

```diff
@@ -268,24 +269,60 @@ uint8_t MaxLevel::GetSpellGainHook(CNWClass *pClass, uint8_t nLevel, uint8_t nSp
     return result;
 }
 
-// If the player is at level 40 or lower we get the spell gain for that class from the normal array,
+// After the server loads 1-40 we populate our map with the values for 41+
+void MaxLevel::LoadSpellKnownTableHook(Services::Hooks::CallType type, CNWClass* pClass, CExoString *pTable)
+{
+    if (type == Services::Hooks::CallType::BEFORE_ORIGINAL)
+        return;
+
+    auto *twoda = new C2DA(CResRef(pTable->CStr()), 1);
```

Comment:
You should just do `C2DA twoda(pTable->CStr(), true);` here, no need to allocate it on the heap, especially since you don't free it when you're done.

Reply:
Same for `LoadSpellGainTableHook()` which I guess I missed last time